src/ray/core_worker/actor_manager.h

```diff
@@ -49,8 +49,12 @@ class DefaultActorCreator : public ActorCreatorInterface {
     auto promise = std::make_shared<std::promise<void>>();
     auto status = gcs_client_->Actors().AsyncRegisterActor(
         task_spec, [promise](const Status &status) { promise->set_value(); });
-    if (status.ok()) {
-      promise->get_future().wait();
+    if (status.ok() && ready_promise->get_future().wait_for(std::chrono::seconds(10)) !=
```

Comment:
How about define a register_actor_timeout_second in ray_config_def.h ?

Reply:
Fixed

Reply:
We use the same pattern for `GetNamedActorHandle`, so I just created a config value for both cases.